modules/smitten/src/defaultEmitter.ts

```diff
@@ -0,0 +1,31 @@
+import {action} from '@action-land/core'
+
+import {EmitValue, ISmitten} from '../index'
+
+import {RootEmitter} from './rootEmitter'
+
+/**
+ * Class for an Emitter that's not at a root level
+ */
+export class DefaultEmitter implements ISmitten {
+  public constructor(
+    public readonly type: string | number,
+    public readonly parent: DefaultEmitter | RootEmitter
+  ) {}
+  public emit = (value: EmitValue): void => {
+    if (this instanceof RootEmitter) {
+      this.emit(value)
+    } else {
+      let current = this.parent
+      let act = action(this.type, value)
```

Comment:
- Tests pass even if `act` is initialized to `value`
  - That seems like it'd be a bug though
- I think a test-case is missing but I'm not able to write a failing test. 🙈